### PR TITLE
Improved CCD calculation algorithm

### PR DESCRIPTION
**Pull request by AndreaCatania**
Improved CCD calculation algorithm

Related to #16113

### CODE REVIEW

File: modules/bullet/rigid_body_bullet.cpp [hunk 1]
```diff
@@ -834,7 +832,7 @@ void RigidBodyBullet::reload_shapes() {
 	btBody->updateInertiaTensor();
 
 	reload_kinematic_shapes();
-
+	set_continuous_collision_detection(btBody->getCcdMotionThreshold() < 10000.0);
```
Comment (akien-mga) [inline modules/bullet/rigid_body_bullet.cpp]: Isn't it a bit risky if that `10000.0` magic threshold ends up being `9999.999123` due to floating point precision errors?

Comment (AndreaCatania) [reply modules/bullet/rigid_body_bullet.cpp]: Fair

**Comment by akien-mga** [comment]
Note that you created this patch on the main godotengine repo instead of your fork: https://github.com/godotengine/godot/tree/AndreaCatania-patch-1

Please try to avoid this in the future, so that we keep the upstream repo with only the active branches.

**Comment by AndreaCatania** [comment]
Sorry I'll do it next time, Thanks

**Comment by mhilbrunner** [comment]
Whats the state of this?

**Comment by AndreaCatania** [comment]
Once compile, can be merged

**Comment by mhilbrunner** [comment]
Merged. Thanks!

**Comment by ZingBlue** [comment]
This might be a silly question to ask but, I am running a 2D game on Godot 3.1.1 and I am having issues with Continuous CD that are really messing with the game's game play, the only file that was changed was `rigid_body_bullet.cpp`, if I remember correctly Bullet is only for 3D. Will this commit somewhat fix my issues with Continuous CD?

**Comment by Calinou** [comment]
@ZingBlue Bullet is only used in 3D, so no. There are other issues related to Continuous CD not working as expected (#9071 in particular).

**Pull request merged**